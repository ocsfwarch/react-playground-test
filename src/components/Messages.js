import React, { Component } from 'react';
import './Messages.css';
import PropTypes from 'prop-types';

export class Messages extends Component {
    constructor(props){
        super(props);
        this.state = {
            name: props.name,
            unread: props.unread
        }
    }

  render() {
    const badge = this.state.unread 
    ? <div className="unread_count">{this.state.unread}</div>
    : null;
    return (
        <div className="messages">
            {this.state.name}
            {badge}
        </div>
    );
  }
}// End class Messages

Messages.propTypes = {
    name: PropTypes.string.isRequired,
    unread: PropTypes.number.isRequired
};


export default Messages
