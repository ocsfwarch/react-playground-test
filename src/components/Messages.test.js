import React from 'react';
import ReactDOM from 'react-dom';
import Renderer from 'react-test-renderer';
import Messages from './Messages';

describe("Messages Component Test Suite", () => {
    // General Smoke Test
    it('Messages renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Messages name="Messages" unread={4} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    // Snapshot test with unread > 0
    it('Messages renders the UI as expected', () => {
        const tree = Renderer
        .create(<Messages name="Messages" unread={4} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // Snapshot test with unread = 0
    it('Messages renders the UI as expected with zero unreads', () => {
        const tree = Renderer
        .create(<Messages name="Messages" unread={0} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
