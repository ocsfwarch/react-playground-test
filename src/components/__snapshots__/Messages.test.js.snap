// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages Component Test Suite Messages renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count"
  >
    4
  </div>
</div>
`;

exports[`Messages Component Test Suite Messages renders the UI as expected with zero unreads 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;

exports[`Messages renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count"
  >
    4
  </div>
</div>
`;

exports[`Messages renders the UI as expected with zero unreads 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
